### PR TITLE
Gun Dryness: Energy gunfire sound pitch now varies based on how much ammo is left. Also guns now have bolt sounds for low ammo. [PORT]

### PR DESCRIPTION
**Pull request by PestoVerde322**
## About The Pull Request

*This is a port of https://github.com/tgstation/tgstation/pull/67549

This Pr adds gun "dryness" sound for both the energy and ballistic guns alike, on top of that, this also add a fuse "CLICK" noise to the energy gun, which was not implemented in the original.

## Why It's Good For The Game

As it says in the original PR over in /tg/, it breaks up the monotony of firearm sounds, makes them actually useful in a gameplay context.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

If i can remember how to record a video with audio, I'll add it soon enough.

</details>

## Changelog
:cl:
soundadd: Energy and Magic gunfire sound pitch now varies based on how much ammo is left.
soundadd: Ballistic and Energy gunfire now has a low-ammo click/fuse sound.
/:cl:

### CODE REVIEW

**Comment by DerpiestBruh** [comment]
> Energy gun effects inspired by Cruelty Squad.

speedmerge.

**Comment by FortSumter** [comment]
Can't wait for three consecutive greenshifts where these sounds aren't heard once...

File: code/modules/projectiles/guns/ballistic/rifle.dm [hunk 1]
```diff
@@ -10,7 +10,7 @@
 	internal_magazine = TRUE
 	fire_sound = "sound/weapons/rifleshot.ogg"
 	fire_sound_volume = 80
-	vary_fire_sound = FALSE
+//	vary_fire_sound = FALSE
```
Comment (PowerfulBacon) [inline code/modules/projectiles/guns/ballistic/rifle.dm]: why comment, just remove

**Review by PowerfulBacon — COMMENTED** [review]
There are a bunch of commented out codes

**Review by ike709 — APPROVED** [review]

**Pull request merged**